openmdao/core/system.py

```diff
@@ -1968,6 +1986,27 @@ def split_list(lst):
                                     (self.pathname, entry))
             return names, patterns, renames
 
+        def update_src_indices(name, key):
+            """
+            Update metadata for promoted inputs that have had src_indices specified.
+
+            Parameters
+            ----------
+            name : str
+                Name of an input variable that may have associated src_indices.
+            key : str or tuple
+                Name, pattern or tuple by which src_indices would have been specified
+                when the input variable was promoted.
+            """
+            if key in self._var_promotes_src_indices:
+                src_indices, flat_src_indices = self._var_promotes_src_indices[key]
+
+                meta = self._var_rel2meta[name]
+                _, _, src_indices = ensure_compatible(name, meta['value'], meta['shape'],
```

Comment:
I didn't see any error checking for the case where src_indices overrides a pre-existing src_indices either from a lower level group.promotes() or from a component add_input().